### PR TITLE
Document nested functions

### PR DESCRIPTION
**Pull request by hawkfish**
I had to read the code to figure them out.

### CODE REVIEW

File: docs/sql/functions/nested.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+layout: docu
+title: Nested Functions
+selected: Documentation/Functions/Nested Functions
+expanded: Functions
+---
+This section describes functions and operators for examining and manipulating nested values.
+
+| Function | Description | Example | Result |
+|:---|:---|:---|:---|
+| `list_extract(`*`list`*`, `*`index`*`)` | Extract the `index`th (0-based) value from the list. | `list_extract(l, 2)` | `6` |
```
Comment (hannesmuehleisen) [inline docs/sql/functions/nested.md:11]: Maybe also mention the  [] syntax that is also supported here

**Comment by hannesmuehleisen** [comment]
Sorry for requiring code reading. We also still need to support nested types in joins and aggregations, but coming.

File: docs/sql/functions/nested.md (hunk 1 above)
Comment (hawkfish) [reply docs/sql/functions/nested.md:11]: And aliases...

**Comment by hawkfish** [comment]
Think it is pretty complete now.  I also found issue #1560.

**Comment by hawkfish** [comment]
@hannesmuehleisen don't apologise!  I do think new features should include documentation changes as part of the commit.

**Pull request merged**